api/bibliotheca.py

```diff
@@ -1323,41 +1325,28 @@ def slice_timespan(self, start, cutoff, increment):
             yield slice_start, slice_cutoff, full_slice
             slice_start = slice_start + increment
 
-    def run_once(self, start, cutoff):
+    def catch_up_from(self, start, cutoff, progress):
```

Comment:
This is a more complicated change, but the end result is the same. This Monitor now fits into the `TimelineMonitor` pattern. We were using the `Timestamp` to keep track of the timestamps on the events that we actually processed, rather than the spans of time we covered, but there was no super-important reason for that, and using `TimelineMonitor` simplifies the code a lot.